proxy/grpcproxy/watch.go

```diff
@@ -203,6 +215,11 @@ func (sws *serverWatchStream) maybeCoalesceWatcher(ws watcherSingle) bool {
 	rid := receiverID{streamID: sws.id, watcherID: ws.w.id}
 	if sws.groups.maybeJoinWatcherSingle(rid, ws) {
 		delete(sws.singles, ws.w.id)
+		// do not add new watchers when stream is closing
+		if sws.inGroups == nil {
```

Comment:
isn't this functionally the same as never setting `sws.inGroups = nil` in the `sws.close` path?